Update jira_issue_worklog table design to avoid API calls to list issues if issue_id is provided in query parameter

mariusgrigaitis opened the pull request:
See https://github.com/turbot/steampipe-plugin-jira/pull/143

This PR applies same fix but to jira_issue_worklog table.

One small note: I noticed that for Get.KeyColumns, AnyColumns is used instead of AllColumns. I believe for successfully getting worklog, both are required as per API design https://developer.atlassian.com/cloud/jira/platform/rest/v2/api-group-issue-worklogs/#api-rest-api-2-issue-issueidorkey-worklog-id-get

Results:

Now:
```
select id from jira_issue_worklog where issue_id = 'JIRA-100';
+-------+
| id    |
+-------+
| 12942 |
+-------+

Time: 278ms. Rows returned: 0. Rows fetched: 1. Hydrate calls: 0.

Scans:
  1) jira_issue_worklog.jira: Time: 225ms. Fetched: 1. Hydrates: 0. Quals: issue_id=JIRA-100.
```

Earlier:

No results (timeout) due to number of issues fetched

ParthaI commented:
Thanks, @mariusgrigaitis, for raising the PR! The changes look great to me.  

I do have a couple of suggestions:  
- Could you please include an example query result in the PR body for better clarity?  
- Could you update the **not found error handling** for the `jira_issue_worklog` table to align with how we handle it for [other tables](https://github.com/turbot/steampipe-plugin-jira/pull/148/files#diff-06088e25404de469e92fd12fcb18c82147fb9574eb8a0cf563f5870fb05e2117R212)?  

Let me know your thoughts. Thanks again! 😊

mariusgrigaitis commented:
@ParthaI I believe I have addressed your points. I have updated PR description as well as made us of isNotFoundError. I have noticed that in your PR (that was merged) it's also not using the predefined function for NotFound, so I've updated that place as well (see https://github.com/turbot/steampipe-plugin-jira/commit/75571c7455d5f15796d29abf6297323de53b7912#diff-06088e25404de469e92fd12fcb18c82147fb9574eb8a0cf563f5870fb05e2117R161-R163 )

mariusgrigaitis commented:
Please check progress on https://github.com/turbot/steampipe-plugin-jira/pull/151 before merging this one. I believe #151 eliminates the problem that this PR tries to solve ;)

misraved commented:
Thanks @mariusgrigaitis for the PR 👍 !!

I think we no longer need this change since https://github.com/turbot/steampipe-plugin-jira/pull/151 has now been merged.
